client/index.ts

```diff
@@ -173,7 +173,7 @@ class OAuth2Client {
 
     const { alg, kid } = decodeJWTPart<{ alg: string; kid: string }>(header);
     const { aud } = decodeJWTPart<{ aud: string }>(payload);
-    let jwks;
+    let jwks = this.jwks;
```

Comment:
I don't get this one. What about:

```ts
const jwks = kid && await this.getJWKS();
```

Maybe it's weird, but I think using again `this.jwks` is what we wanted to avoid ^^

Reply:
Yep, I agree that this is what we wanted to avoid. I like your proposal 🙂